src/signals/IncidentMap/components/IncidentLayer/IncidentLayer.tsx

```diff
@@ -1,81 +1,104 @@
 /* SPDX-License-Identifier: MPL-2.0 */
 /* Copyright (C) 2022 Gemeente Amsterdam */
-import { useEffect, useState } from 'react'
+import { useEffect, useRef, useState } from 'react'
 
+import type { FeatureCollection, Point } from 'geojson'
 import L from 'leaflet'
 
 import MarkerCluster from 'components/MarkerCluster'
-import { dynamicIcon } from 'shared/services/configuration/map-markers'
-import { featureToCoordinates } from 'shared/services/map-location'
+import {
+  dynamicIcon,
+  selectedMarkerIcon,
+} from 'shared/services/configuration/map-markers'
 import type { Bbox } from 'signals/incident/components/form/MapSelectors/hooks/useBoundingBox'
 import useBoundingBox from 'signals/incident/components/form/MapSelectors/hooks/useBoundingBox'
 
-import type { Point, Incident } from '../../types'
+import type { Incident } from '../../types'
 
 const clusterLayerOptions = {
   zoomToBoundsOnClick: true,
   chunkedLoading: true,
 }
+const emptyFeatureCollection = (): FeatureCollection => ({
+  type: 'FeatureCollection',
+  features: [],
+})
 
 interface Props {
+  handleIncidentSelect: (incident: Incident) => void
   incidents?: Incident[]
   passBbox(bbox: Bbox): void
-  resetMarkerIcons: () => void
-  handleIncidentSelect: (incident?: Incident) => void
-  handleCloseDetailPanel: () => void
+  resetMarkerIcon: () => void
+  selectedMarkerRef: React.MutableRefObject<L.Marker<Incident> | undefined>
 }
 
 /* istanbul ignore next */
 export const IncidentLayer = ({
+  handleIncidentSelect,
   incidents,
   passBbox,
-  resetMarkerIcons,
-  handleIncidentSelect,
-  handleCloseDetailPanel,
+  resetMarkerIcon,
+  selectedMarkerRef,
 }: Props) => {
   const [layerInstance, setLayerInstance] = useState<L.GeoJSON<Point>>()
+  const activeLayer = useRef<L.GeoJSON>()
   const bbox = useBoundingBox()
 
   useEffect(() => {
     if (bbox) {
       passBbox(bbox)
+      resetMarkerIcon()
     }
-  }, [bbox, passBbox])
+  }, [bbox, passBbox, resetMarkerIcon])
 
   useEffect(() => {
     if (!incidents || !layerInstance) return
+    activeLayer.current?.remove()
+    const fc = emptyFeatureCollection()
 
-    incidents.forEach((incident) => {
-      const latlng = featureToCoordinates(incident.geometry)
-      const { category, icon: categoryIcon } = incident.properties
+    fc.features = incidents
+    layerInstance.clearLayers()
 
-      const marker = L.marker(latlng, {
-        icon: dynamicIcon(categoryIcon),
-        alt: category.name,
-        keyboard: false,
-      })
+    const layer = L.geoJSON(fc, {
+      onEachFeature: (feature: Incident, layer: L.Layer) => {
+        layer.on('click', (e: { target: L.Marker<Incident> }) => {
```

Comment:
e: LeafletEvent

Reply:
Then target becomes `any`, which is not what we want I think.